test_agent/rust/src/testagent.rs

```diff
@@ -243,14 +243,14 @@ impl SocketTestAgent {
 
             let status = match json_str_ref {
                 constants::SEND_COMMAND => {
-                    self.handle_send_command(&utransport, json_data_value).await
+                    self.handle_send_command(&*utransport, json_data_value).await
```

Comment:
Only question I have is, I have added the Box implementation to pass into this method, but for the handle methods, I just end up dereferencing the variable anyway... is that acceptable?

Reply:
Yeah, this is okay! We've passed in the `Box<dyn UTransport>` to this function, transferring ownership to this function of the heap allocated memory.

We can then feel free to pass out `&dyn UTransport`, which is saying we'll let the functions take a _non-exclusive_ reference to the trait object.

Reply:
cool!